packages/ui/src/views/layouts/content-layout-with-sidebar.tsx

```diff
@@ -0,0 +1,75 @@
+import { FC, ReactNode } from 'react'
+import { NavLink } from 'react-router-dom'
+
+import { NavbarSkeleton, ScrollArea } from '@/components'
+import { cn } from '@utils/cn'
+
+export interface SidebarMenuItemSubItem {
+  id: string | number
+  title: string
+  to: string
+  description?: string
+}
+
+export interface SidebarMenuItem {
+  groupId: number | string
+  title?: string
+  items: SidebarMenuItemSubItem[]
+}
+
+export interface ContentLayoutWithSidebarProps {
+  children: ReactNode
+  sidebarMenu: SidebarMenuItem[]
+  sidebarOffsetTop?: number
+  // You need to pass the padding-top value, which is the initial offset of the sidebar from the top of the visible area.
+  // This is required to ensure that scrolling within the block goes beyond the boundary
+  // while the block maintains the correct top padding according to the design.
+  sidebarViewportClassName?: string
+}
+
+export const ContentLayoutWithSidebar: FC<ContentLayoutWithSidebarProps> = ({
+  children,
+  sidebarMenu,
+  sidebarOffsetTop = 0,
+  sidebarViewportClassName
+}) => {
+  return (
+    <div className="relative mx-auto flex w-[1072px] items-start gap-x-20">
```

Comment:
```suggestion
    <div className="relative mx-auto flex w-full max-w-[1072px] items-start gap-x-20">
```


In this case, I would approach it differently to make the container more flexible and not so rigid. What do you think about this?

Reply:
This is a deliberately restricted wrapper by design; this layout is specifically for such a grid. The sidebar menu is 220px, and the content should fit into the remaining width.